### PR TITLE
changes suggested by dependabot

### PR DESCRIPTION
**Pull request by wxtim**
These changes replace dependabot PR's:
#856 
#857 
#870 
#872 

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Already covered by existing tests.
- [x] No change log entry required (Dependabot change).
- [x] No documentation update required.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -36,6 +36,12 @@ Vue.prototype.$eventBus = mitt()
 Vue.config.productionTip = false
 
 /* eslint-disable no-new */
+/*
+This is where the main app is created:
+- What are all the objects (?) being fed in?
+- What is render (h) doing?
+- What is ``$mount('#app')``?
+*/
```
Comment (wxtim) [inline src/main.js:44]: @oliver-sanders  are these valid questions to leave in the docs?
```suggestion
```

File: src/components/cylc/cylcObject/Menu.vue [hunk 2]
```diff
@@ -43,6 +43,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
             <v-icon :disabled=!authorised large>{{ mutation._icon }}</v-icon>
           </v-list-item-avatar>
           <v-list-item-content>
+            <!-- Is this safe against XSS scripting? -->
```
Comment (wxtim) [inline src/components/cylc/cylcObject/Menu.vue:46]: Valid comment?
```suggestion
```

Comment (oliver-sanders) [reply src/components/cylc/cylcObject/Menu.vue:46]: Your comment?

Comment (wxtim) [reply src/components/cylc/cylcObject/Menu.vue:46]: Automatically added by build process

File: src/main.js (hunk 1 above)
Comment (oliver-sanders) [reply src/main.js:44]: * The "router" is the bit that translates URLs into views (e.g. `workflows` goes to the Workflow view)
* The "store" is the VueX store instance.
* "Vuetify" is a component library (we use it for buttons etc).
* "render" is the Vue render function.
* "$mount('#app')" causes the vue instance to be "mounted" on the HTML element with the ID "app"

Comment (wxtim) [reply src/main.js:44]: Again, these were added by `yarn install`. Should I edit to actually answer the questions?

File: src/components/cylc/cylcObject/Menu.vue (hunk 2 above)
Comment (oliver-sanders) [reply src/components/cylc/cylcObject/Menu.vue:46]: ???

Comment (wxtim) [reply src/components/cylc/cylcObject/Menu.vue:46]: When I ran `yarn install` it added this line.

File: src/main.js (hunk 1 above)
Comment (oliver-sanders) [reply src/main.js:44]: Delete.

File: src/components/cylc/cylcObject/Menu.vue (hunk 2 above)
Comment (oliver-sanders) [reply src/components/cylc/cylcObject/Menu.vue:46]: Will raise as an issue, may be able to do this better, otherwise delete.

Comment (oliver-sanders) [reply src/components/cylc/cylcObject/Menu.vue:46]: https://github.com/cylc/cylc-ui/issues/875

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**